stem/model.py

```diff
@@ -191,6 +172,186 @@ def generate_straight_track(self, sleeper_distance: float, n_sleepers: int, rail
+                                         origin_point: Sequence[float], extended_soil_parameters: ElasticSpringDamper,
+                                         length_extended_soil: float, direction_vector: Sequence[float], name: str):
+        """
+        Generates a track geometry. With rail, rail-pads and sleepers as mass elements. Sleepers are placed at the
+        bottom of the track with a distance of sleeper_distance between them. The sleepers are connected to the rail
+        with rail-pads with a thickness of rail_pad_thickness. The track is generated in the direction of the
+        direction_vector starting from the origin_point. The track can only move in the vertical direction.
+
+        Args:
+            - sleeper_distance (float): distance between sleepers
+            - n_sleepers (int): number of sleepers
+            - rail_parameters (:class:`stem.structural_material.EulerBeam`): rail parameters
+            - sleeper_parameters (:class:`stem.structural_material.NodalConcentrated`): sleeper parameters
+            - rail_pad_parameters (:class:`stem.structural_material.ElasticSpringDamper`): rail pad parameters
+            - rail_pad_thickness (float): thickness of the rail pad
+            - origin_point (Sequence[float]): origin point of the track
+            - direction_vector (Sequence[float]): direction vector of the track
+            - extended_soil_parameters: (:class:`stem.structural_material.ElasticSpringDamper`): soil equivalent
+            parameters
+            - length_extended_soil (float): length of the 1D soil equivalent
+            - name (str): name of the track
+        """
+        self.__define_straight_track(sleeper_distance, n_sleepers, rail_parameters, sleeper_parameters,
+                                     rail_pad_parameters, rail_pad_thickness, origin_point, direction_vector, name)
+        self.__generate_extended_rail_part(extended_soil_parameters, name, length_extended_soil)
+
+    def __generate_extended_rail_part(self, extended_soil_parameters: ElasticSpringDamper, name: str,
+                                      length_extended_soil: float):
+        """
+        Generates the soil equivalent of the rail part. The soil equivalent is a mass element that represents the
+        soil below the rail. The soil equivalent is connected to the rail with rail-pads. The bottom of the soil
+        equivalent is fixed in all directions. While the soil equivelent can only move in the vertical direction.
+
+        Args:
+            - extended_soil_parameters: (:class:`stem.structural_material.ElasticSpringDamper`): soil equivalent
+            parameters
+            - name (str): name of the track
+            - length_extended_soil (float): length of the 1D soil equivalent
+        """
+
+        soil_equivalent_name = f"soil_equivalent_{name}"
+        sleeper_name = f"sleeper_{name}"
+        soil_equivalent_bottom_length = length_extended_soil
```

Comment:
the length follows from the bounding box of the soil

Reply:
I don't think it matters the depth of the equivalent soil domain should be set by the user right?

Reply:
ok